src/braket/default_simulator/openqasm/data_manipulation.py

```diff
@@ -0,0 +1,656 @@
+def _(into: IntType, variable: LiteralType) -> IntegerLiteral:
+    """Cast to int with overflow warnings"""
+    limit = 2 ** (into.size.value - 1)
+    value = int(np.sign(variable.value) * (np.abs(int(variable.value)) % limit))
+    if value != variable.value:
+        warnings.warn(f"Integer overflow for value {variable.value} and size {into.size.value}.")
+    return IntegerLiteral(value)
+
+
+@cast_to.register
+def _(into: UintType, variable: LiteralType) -> IntegerLiteral:
+    """Cast to uint with overflow warnings. Bit registers can be cast to uint."""
+    if isinstance(variable, ArrayLiteral):
+        return IntegerLiteral(int("".join("01"[x.value] for x in variable.values), base=2))
+    limit = 2**into.size.value
+    value = int(variable.value) % limit
+    if value != variable.value:
+        warnings.warn(
+            f"Unsigned integer overflow for value {variable.value} and size {into.size.value}."
+        )
+    return IntegerLiteral(value)
+
+
+@cast_to.register
+def _(into: FloatType, variable: LiteralType) -> FloatLiteral:
+    """Cast to float"""
+    if into.size.value not in (16, 32, 64, 128):
+        raise ValueError("Float size must be one of {16, 32, 64, 128}.")
+    value = float(np.array(variable.value, dtype=np.dtype(f"float{into.size.value}")))
+    return FloatLiteral(value)
+
+
+@cast_to.register
+def _(into: ArrayType, variable: Union[ArrayLiteral, DiscreteSet]) -> ArrayLiteral:
+    """Cast to Array and enforce dimensions"""
+    if len(variable.values) != into.dimensions[0].value:
+        raise ValueError(
+            f"Size mismatch between dimension of size {into.dimensions[0].value} "
+            f"and values length {len(variable.values)}"
+        )
+    subtype = (
+        ArrayType(into.base_type, into.dimensions[1:])
+        if len(into.dimensions) > 1
+        else into.base_type
+    )
+    return ArrayLiteral([cast_to(subtype, v) for v in variable.values])
+
+
+def is_literal(expression: Expression) -> bool:
+    return isinstance(
+        expression,
+        (
+            BooleanLiteral,
+            IntegerLiteral,
+            FloatLiteral,
+            BitstringLiteral,
+            ArrayLiteral,
+        ),
+    )
+
+
+def convert_string_to_bool_array(bit_string: BitstringLiteral) -> ArrayLiteral:
+    """Convert BitstringLiteral to Boolean ArrayLiteral"""
+    return ArrayLiteral(
+        [BooleanLiteral(x == "1") for x in np.binary_repr(bit_string.value, bit_string.width)]
+    )
+
+
+def convert_bool_array_to_string(bit_string: ArrayLiteral) -> str:
+    """Convert Boolean ArrayLiteral into a binary string"""
+    return "".join(("1" if x.value else "0") for x in bit_string.values)
+
+
+def is_none_like(value: Any) -> bool:
+    """Returns whether value is None or an Array of Nones"""
+    if isinstance(value, ArrayLiteral):
+        return all(is_none_like(v) for v in value.values)
+    return value is None
+
+
+"""
+Helper functions for working with indexed values
```

Comment:
I can do that if you think it would be cleaner, I just went with one file for all helper methods, but open to shaking it up. Are you asking if I can move these functions to another file without creating circular dependencies?

Reply:
Yes. I mean I know we can deal with circular dependencies in either case, but if there is a clean split between these it makes it all the more obvious we should split them. 

These kind of files get to be unwieldily quickly and if they (at least) have a name that corresponds to the functionality it's a warning sign when you start adding functions that don't fit the name (kind of similar to classes and putting non-class-specific stuff in the class).